File: reporting-winforms-master-detail-subreport-23.2.3-/CS/dxSampleMasterDetailSubreport/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace dxSample {
    public partial class Form1 : DevExpress.XtraBars.ToolbarForm.ToolbarForm {
        public Form1() {
            InitializeComponent();
        }

        private void barButtonItem1_ItemClick(object sender, DevExpress.XtraBars.ItemClickEventArgs e) {
            barButtonItem2.Down = false;
            var report = new MasterReport();
            report.CreateDocument();
            this.documentViewer1.DocumentSource = report;
        }

        private void barButtonItem2_ItemClick(object sender, DevExpress.XtraBars.ItemClickEventArgs e) {
            barButtonItem1.Down = false;
            var report = new MasterReportOne();
            report.DataSource = CategoryProductDataSource.GetData();
            report.CreateDocument();
            this.documentViewer1.DocumentSource = report;
        }


    }
}

File: reporting-winforms-master-detail-subreport-23.2.3-/VB/dxSampleMasterDetailSubreport/Form1.vb
Imports System.ComponentModel
Imports System.Text


Partial Public Class Form1
    Public Sub New()
        InitializeComponent()
    End Sub

    Private Sub BarButtonItem1_ItemClick(sender As Object, e As DevExpress.XtraBars.ItemClickEventArgs) Handles BarButtonItem1.ItemClick
        BarButtonItem2.Down = False
        Dim report As New MasterReport()
        report.CreateDocument()
        DocumentViewer1.DocumentSource = report
    End Sub

    Private Sub BarButtonItem2_ItemClick(sender As Object, e As DevExpress.XtraBars.ItemClickEventArgs) Handles BarButtonItem2.ItemClick
        BarButtonItem1.Down = False
        Dim report As New MasterReportOne()
        report.DataSource = CategoryProductDataSource.GetData()
        report.CreateDocument()
        DocumentViewer1.DocumentSource = report
    End Sub
End Class
